src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -30,42 +36,95 @@ import { fmt_to_fragments } from 'utils.jsx';
 import { NeedsShutdownAlert } from '../common/needsShutdown.jsx';
 import { domainReplaceSpice } from '../../libvirtApi/domain.js';
 
+import './vmReplaceSpiceDialog.scss';
+
 const _ = cockpit.gettext;
 
-export const ReplaceSpiceDialog = ({ vmName, vmId, connectionName, vmRunning }) => {
+export const ReplaceSpiceDialog = ({ vm, vms }) => {
+    const spiceVMs = vms?.filter(vm => vm.inactiveXML?.hasSpice);
+    // sort spiceVMs by name, put the current VM first
+    spiceVMs?.sort((a, b) => (a.name === vm.name) ? -1 : (b.name === vm.name) ? 1 : a.name.localeCompare(b.name));
```

Comment:
They are guaranteed *not* to be unique across different connections; you can easily have a "test1" on system and session. Indeed this could be `a === vm` and `b === vm`, I'll test if that works.